### PR TITLE
Update wellbore properties

### LINKED ISSUE #456: User story:  WELLBORE Backend - Update attributes in property window
As a Data Manager, I want to be able to right click on a WELLBORE, and update the attributes in the property window.

NB: This backlog Item only covers single instance update, not batch update of multiple object instances):

EDIT, NO BATCH, NO ENUM:
nameWellbore, nameWell, number, suffixAPI, numGovt, dTimKickoff, md, tvd, mdKickoff,tvdKickoff, mdPlanned, tvdPlanned, mdDubSeaPlanned, tvdSubseaPlanned, dayTarget

EDIT, BATCH, OPEN ENUM:
statusWellbore, purposeWellbore, typeWellbore, shape

### Acceptance criteria ###
- [ ] Ability to right click on a WELLBORE Object instance, and update the WELLBORE attributes listed above.

### Resources ###

-  Lead resource: KDI
-  Contributors: KDI

### PR DESCRIPTION
**Pull request by kdi-chandra**
# Request Template Witsml Explorer
Thank you for contributing to WITSML Explorer!
Before submitting this PR, please fill in the following template.

## Fixes
This pull request fixes #456

## Description
This pull request has API changes for updating wellbore properties

## Type of change
_List the types of change. Remove from the list any points which are not necessary_

* New feature (non-breaking change which adds functionality)
* Enhancement of existing functionality

## Impacted Areas in Application
_List general components of the application that this PR will affect:_

API/Backend
Witsml
WitsmlExplorer.Api

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [x] PR is related to an issue

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [x] New code is covered by passing tests

## Further comments
NA

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Query/WellboreQueries.cs [hunk 1]
```diff
@@ -51,16 +53,59 @@ public static WitsmlWellbores GetWitsmlWellboreByUid(string wellUid, string well
...
+                witsmlWellbore.SuffixAPI = wellbore.SuffixAPI;
+
+            if (!string.IsNullOrEmpty(wellbore.NumGovt))
+                witsmlWellbore.NumGovt = wellbore.NumGovt;
+
+            if (wellbore.DTimeKickoff != null)
+                witsmlWellbore.DTimKickoff = ((DateTime) wellbore.DTimeKickoff).ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.fffZ");
+
+            if (wellbore.Md != null)
+                witsmlWellbore.Md = new WitsmlMeasuredDepthCoord { Uom = wellbore.Md.Uom, Value = wellbore.Md.Value.ToString() };
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Query/WellboreQueries.cs]: Consider specifying string culture explicitly

File: Src/WitsmlExplorer.Api/Workers/ModifyWellboreWorker.cs [hunk 2]
```diff
@@ -55,7 +47,17 @@ public async Task<(WorkerResult, RefreshAction)> Execute(ModifyWellboreJob job)
 
         private void Verify(Wellbore wellbore)
         {
-            if (string.IsNullOrEmpty(wellbore.Name)) throw new InvalidOperationException($"{nameof(wellbore.Name)} cannot be empty");
+            if (string.IsNullOrEmpty(wellbore.Uid) || string.IsNullOrEmpty(wellbore.WellUid)) throw new InvalidOperationException($"uid/wellUid cannot be empty");
+            if (wellbore.Name == string.Empty) throw new InvalidOperationException($"{nameof(wellbore.Name)} cannot be empty");
+            if (wellbore.Md != null && string.IsNullOrEmpty(wellbore.Md.Uom)) throw new InvalidOperationException($"unit of measure for Md cannot be empty");
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Workers/ModifyWellboreWorker.cs]: Consider using `nameof{variable}`.

File: Tests/WitsmlExplorer.Api.Tests/Workers/ModifyWellboreWorkerTests.cs [hunk 3]
```diff
@@ -58,6 +58,294 @@ public async Task RenameWellbore_EmptyName_ThrowsException()
             witsmlClient.Verify(client => client.UpdateInStoreAsync(It.IsAny<WitsmlWellbores>()), Times.Never);
         }
 
+        [Fact]
+        public async Task Update_Md()
+        {
+            var expectedValue = 10;
```
Comment (HakonHesselberg) [inline Tests/WitsmlExplorer.Api.Tests/Workers/ModifyWellboreWorkerTests.cs]: Consider changing to const.

**Review by HakonHesselberg — APPROVED** [review]
Looks good to me, tests pass.

- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

File: Src/WitsmlExplorer.Api/Query/WellboreQueries.cs (hunk 1 above)
Comment (kdi-chandra) [reply Src/WitsmlExplorer.Api/Query/WellboreQueries.cs]: Review comments incorporated

File: Src/WitsmlExplorer.Api/Workers/ModifyWellboreWorker.cs (hunk 2 above)
Comment (kdi-chandra) [reply Src/WitsmlExplorer.Api/Workers/ModifyWellboreWorker.cs]: Review comments incorporated

File: Tests/WitsmlExplorer.Api.Tests/Workers/ModifyWellboreWorkerTests.cs (hunk 3 above)
Comment (kdi-chandra) [reply Tests/WitsmlExplorer.Api.Tests/Workers/ModifyWellboreWorkerTests.cs]: Review comments incorporated

**Comment by kdi-chandra** [comment]
Incorporated the changes for review comments, merging the PR

**Pull request merged**